### PR TITLE
Add string palindrome algorithm

### LINKED ISSUE #27: Implementations of checking if a string is palindrome
Good evening Sir, I would like to add 3 implementations of checking if a string is palindrome, including implementation using for loop, recursion, and reversed() function. Do you think it will be useful?

### PR DESCRIPTION
**Pull request by nktssk**
Fixes #27

Implemented algorithm.

### CODE REVIEW

**Comment by AnupKumarPanwar** [comment]
Can you create 3 different files with proper names for the 3 different approaches?

File: algorithms/palindrome/palindrome.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// A palindrome is a string that reads the same forwards and backwards.
+//
+// Examples: "level", "radar", "madam", "A man, a plan, a canal: Panama".
+
+struct PalindromeSolutionFirst {
+    
+    /// Using the `reverse()` method to reverse the string and comparing it with the original. Only include letters and numbers.
+    /// - Complexity: O(n), with allocating O(n) space.
+    func isPalindrome(_ input: String) -> Bool {
```
Comment (5sw) [inline algorithms/palindrome/palindrome.swift:9]: Doesn’t really make sense to have this as an instance method on an empty struct. I would suggest an extension method on String. If you really want to namespace this make this a `static func` on an empty enum.

File: algorithms/palindrome/palindrome.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+struct PalindromeSolutionSecond {
+    
+    /// Iteratively comparing characters from the beginning and end of the string. Only include letters and numbers.
+    /// - Complexity: O(n), without allocating new space.
+    func isPalindrome(_ input: String) -> Bool {
+        var leftIndex = input.startIndex
+        var rightIndex = input.index(before: input.endIndex)
+        
+        while leftIndex < rightIndex {
+            if !input[leftIndex].isLetter && !input[leftIndex].isNumber {
```
Comment (5sw) [inline algorithms/palindrome/palindrome.swift:24]: I’d use a `guard` statement here.

File: algorithms/palindrome/palindrome.swift (hunk 1 above)
Comment (nktssk) [reply algorithms/palindrome/palindrome.swift]: fixed

File: algorithms/palindrome/palindrome.swift (hunk 2 above)
Comment (nktssk) [reply algorithms/palindrome/palindrome.swift]: fixed

**Review by carloshmartins — APPROVED** [review]
LGTM

**Pull request merged**